packages/pros/src/adi/encoder.rs

```diff
@@ -0,0 +1,62 @@
+use crate::adi::{
+    AdiError,
+    AdiSlot
+};
+
+use pros_sys::{
+    PROS_ERR,
+    adi_encoder_t
+};
+
+use crate::error::bail_on;
+
+pub struct AdiEncoder {
+    port_top: u8,
+    port_bottom: u8,
+    reverse: bool,
+    reference: adi_encoder_t
```

Comment:
This is a nitpick (sorry!) but I might name this something like `raw`, since "reference" might refer to the pointer-type of reference. It's weird that the PROS API follows this pattern of passing structs for just one device.